code/modules/reagents/reagent_containers/syringes.dm

```diff
@@ -225,26 +228,48 @@
 		return
 
 	var/rounded_vol
-	if(reagents && reagents.total_volume)
-		rounded_vol = CLAMP(round((reagents.total_volume / volume * 15),5), 1, 15)
-		var/image/filling_overlay = mutable_appearance('icons/obj/reagentfillings.dmi', "syringe[rounded_vol]")
-		filling_overlay.color = reagents.get_color()
-		add_overlay(filling_overlay)
-	else
-		rounded_vol = 0
-
-	icon_state = "[rounded_vol]"
-	item_state = "syringe_[rounded_vol]"
-
-	if(ismob(loc))
-		var/injoverlay
-		switch(mode)
-			if (SYRINGE_DRAW)
-				injoverlay = "draw"
-			if (SYRINGE_INJECT)
-				injoverlay = "inject"
-		add_overlay(injoverlay)
-		update_wear_icon()
+	if( name == "syringe")
```

Comment:
This should not check the name as there are srynges with different names.

Reply:
You should never check for names anyway. It's really bad practice. Use ``istype`` proc.